internal/controllers/topology/cluster/current_state.go

```diff
@@ -72,8 +74,12 @@ func (r *Reconciler) getCurrentState(ctx context.Context, s *scope.Scope) (*scop
 
 // getCurrentInfrastructureClusterState looks for the state of the InfrastructureCluster. If a reference is set but not
 // found, either from an error or the object not being found, an error is thrown.
-func (r *Reconciler) getCurrentInfrastructureClusterState(ctx context.Context, cluster *clusterv1.Cluster) (*unstructured.Unstructured, error) {
-	infra, err := r.getReference(ctx, cluster.Spec.InfrastructureRef)
+func (r *Reconciler) getCurrentInfrastructureClusterState(ctx context.Context, blueprint *scope.ClusterBlueprint, cluster *clusterv1.Cluster) (*unstructured.Unstructured, error) {
```

Comment:
nit: accept only blueprint.InfrastructureClusterTemplate instead of the full ClusterBlueprint
(same for other methods, let's scope down input parameters when possible)